RELEASE_PROCESS.md

```diff
@@ -0,0 +1,51 @@
+# Release Process
+
+This document contains all the necessary information about out release process.
+
+## Release branch
+
+For each release we create a separate release branch.
```

Comment:
Another issue is that master wouldn't reflect the most recent changes, and we'd loose some of our edge users who are using master (which is assumed to include the un-released edge). Sure some of them would figure it out, but I'd guess that most would stay in master.